.github/workflows/main.yml

```diff
@@ -24,7 +24,7 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@stable
```

Comment:
Instead of `stable`, this needs to use the matrix variable `${{ matrix.toolchain }}`.

Thanks for the PR.

Reply:
According to the docs for `rust-toolchain`, I think if we set the `rev` to `master` it will use the toolchain key.

```
Rustup toolchain specifier e.g. stable, nightly, 1.42.0, nightly-2022-01-01. Important: the default is to match the @rev as described above. When passing an explicit toolchain as an input instead of @rev, you'll want to use "dtolnay/rust-toolchain@master" as the revision of the action. 
```